pandas/_libs/src/ujson/python/date_conversions.c

```diff
@@ -90,6 +90,10 @@ char *PyDateTimeToIso(PyObject *obj, NPY_DATETIMEUNIT base,
 
     *len = (size_t)get_datetime_iso_8601_strlen(0, base);
     char *result = PyObject_Malloc(*len);
+    if (result == NULL) {
```

Comment:
I think the idiomatic approach here is to just do `return PyErr_NoMemory()`

https://docs.python.org/3/c-api/exceptions.html#c.PyErr_NoMemory

example from CPython source:

https://github.com/python/cpython/blob/9f2c479eaf7d922746ef2f3c85b5c781757686b1/Objects/unicodeobject.c#L1188